### DISCUSSION TITLE
Web3Modal is not detecting the injected Frontier extension in the browser

### DISCUSSION THREAD

**Discussion by punithbm** [WalletConnect/walletconnect-monorepo#2520] [Developer Support] [answered]
I have updated the required configuration for detecting the injected frontier wallet in the window under web3 modal. But still it is not getting detected. You can find the configuration screenshot below, Please check and let me know if anything is missing 

![Screenshot 2023-06-01 at 10 16 02 AM](https://github.com/WalletConnect/walletconnect-monorepo/assets/13044958/183e546f-7859-4049-82df-e10345c10e21)
![Screenshot 2023-06-01 at 10 15 37 AM](https://github.com/WalletConnect/walletconnect-monorepo/assets/13044958/0f9fcf81-5e32-408a-a0a4-ffc350d8004b)
![Screenshot 2023-06-01 at 10 15 18 AM](https://github.com/WalletConnect/walletconnect-monorepo/assets/13044958/421857ef-56be-4e00-902c-97502c1b7b87)

**Comment by punithbm** [comment]
This is how is showing up in the web3modal when trying to get it connected. Desktop wallet itself not getting detected even though wallet extension is present in the browser. Always asking to scan and open the mobile app

![Screenshot 2023-06-01 at 10 14 29 AM](https://github.com/WalletConnect/walletconnect-monorepo/assets/13044958/ee496f26-6770-45e7-a109-151a7460fdd1)

**Comment by finessevanes** [comment]
@JeroldG could you give some insight 🙏

**Comment by finessevanes** [comment]
@punithbm I just want to confrim the following flags are intentional:
`window.ethereum.frontier.ethereum.isFrontier`
`window.cosmos.frontier.cosmos`
`window.solana.frontier.solana`

**Reply by punithbm** [reply]
Oh, it should be window.frontier.ethereum

I think custom providers like window.frontier is not supported? Then we should use window.ethereum.isFrontier 
And I should add just isFrontier in eip155 field?

**Reply by punithbm** [reply]
Updating like this 👇  should be fine ?
![Screenshot 2023-06-02 at 2 31 10 PM](https://github.com/WalletConnect/walletconnect-monorepo/assets/13044958/776e0c6b-3f3c-486c-b62e-97e7d90e053e)

cause we are injecting both `window.frontier.ethereum.isFrontier` & `window.ethereum.isFrontier`

**Reply by finessevanes** [reply, marked as answer]
>Updating like this 👇 should be fine ?

Yep! I approved your submission. Let me know if you're able to see it now.

**Reply by punithbm** [reply]
Yes working fine. Thank you
